tests/LanguageServerRename/Bridge/Unit/ChainRenamerTest.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+namespace Phpactor\Extension\LanguageServerRename\Tests\Bridge\Unit;
+
+use Generator;
+use PHPUnit\Framework\TestCase;
+use Phpactor\Extension\LanguageServerRename\Bridge\ChainRenamer;
+use Phpactor\Extension\LanguageServerRename\Model\RenameResult;
+use Phpactor\Extension\LanguageServerRename\Model\Renamer;
+use Phpactor\TextDocument\ByteOffset;
+use Phpactor\TextDocument\ByteOffsetRange;
+use Phpactor\TextDocument\TextDocument;
+use Phpactor\TextDocument\TextDocumentBuilder;
+use Phpactor\TextDocument\TextDocumentUri;
+use Phpactor\TextDocument\TextEdits;
+use function iterator_to_array;
+
+class ChainRenamerTest extends TestCase
+{
+    public function testGetRenameRange_OneMatch(): void
+    {
+        $renamer1 = new class() implements Renamer {
+            public function getRenameRange(TextDocument $textDocument, ByteOffset $offset): ?ByteOffsetRange
+            {
+                return new ByteOffsetRange(ByteOffset::fromInt(0), ByteOffset::fromInt(1));
+            }
+            
+            public function rename(TextDocument $textDocument, ByteOffset $offset, string $newName): Generator
+            {
+                return;
+                yield;
+            }
+        };
```

Comment:
Rather than using anonymous classes, could we implement a pre defined renamer? (or similar name, TestRenamer, StubRenamer etc)

```
new PredefinedRenamer($byteOffsetRange, $renameResults)
```

This could repalce the stub Member and Variable renamers and potentially make the tests easier?

Possibly you could also implement a builder pattern:

```
PredefinedResult::fromRange(ByteOffsetRange(...))->withResult(...)->withResult(...)
```

Reply:
Sounds good.